duckduckgo/Android
# ADS: SectionHeaderListItem component

Task/Issue URL: https://app.asana.com/0/0/1203197728681129/f

### Description

Create `SectionHeaderListItem` which includes the optional overflow menu as trailing icon. It will replace all `HeaderSectionTextView` in migration step for ADS.

### Steps to test this PR

- Install from this branch
- Go to Settings > App Components Design Preview
- Tap on List items tab
- [ ] Check Section Header use cases match new designs

### UI changes
| Before  | After |
| ------ | ----- |
![Screenshot_20221101-133747_DuckDuckGo](https://user-images.githubusercontent.com/20798495/199247948-30accca8-8ac8-4f5d-af69-be8af0c69f67.jpg)|![Screenshot_20221101-114111_DuckDuckGo](https://user-images.githubusercontent.com/20798495/199248013-9500ccc3-c771-4f23-91a5-360857326efc.jpg)|
